components/schema-migrator/migrations/201908141300_initialize_schema.up.sql

```diff
@@ -0,0 +1,223 @@
+    'UNKNOWN',
+    'READY',
+    'FAILED'
+);
+
+CREATE TABLE applications (
+    id uuid PRIMARY KEY,
+    tenant_id uuid NOT NULL,
+    name varchar(36) NOT NULL,
+    description text,
+    status_condition application_status_condition DEFAULT 'INITIAL' ::application_status_condition NOT NULL,
+    status_timestamp timestamp NOT NULL,
+    healthcheck_url varchar(256)
+);
+
+ALTER TABLE applications
+    ADD CONSTRAINT application_id_name_unique UNIQUE (tenant_id, name);
+
+CREATE INDEX ON applications (tenant_id);
+
+-- Webhook
+
+CREATE TYPE webhook_type AS ENUM (
+    'CONFIGURATION_CHANGED'
+);
+
+CREATE TABLE webhooks (
+    id uuid PRIMARY KEY,
+    tenant_id uuid NOT NULL,
+    app_id uuid REFERENCES applications (id) ON DELETE CASCADE NOT NULL,
+    url varchar(256) NOT NULL,
+    type webhook_type NOT NULL,
+    auth jsonb
+);
+
+CREATE INDEX ON webhooks (tenant_id);
+
+-- API Definition
+
+CREATE TYPE api_spec_format AS ENUM (
+    'YAML',
+    'JSON'
+);
+
+CREATE TYPE api_spec_type AS ENUM (
+    'ODATA',
+    'OPEN_API'
+);
+
+CREATE TABLE api_definitions (
+    id uuid PRIMARY KEY,
+    tenant_id uuid NOT NULL,
+    app_id uuid REFERENCES applications (id) ON DELETE CASCADE NOT NULL,
+    name varchar(256) NOT NULL,
+    description text,
+    group_name varchar(256),
+    target_url varchar(256) NOT NULL,
+    spec_data text,
+    spec_format api_spec_format,
+    spec_type api_spec_type,
+    default_auth jsonb,
+    version_value varchar(256),
+    version_deprecated bool,
+    version_deprecated_since varchar(256),
+    version_for_removal bool
+);
+
+CREATE INDEX ON api_definitions (tenant_id);
+
+-- Event API Definition
+
+CREATE TYPE event_api_spec_format AS ENUM (
+    'YAML',
+    'JSON'
+);
+
+CREATE TYPE event_api_spec_type AS ENUM (
+    'ASYNC_API'
+);
+
+CREATE TABLE event_api_definitions (
+    id uuid PRIMARY KEY,
+    tenant_id uuid NOT NULL,
+    app_id uuid REFERENCES applications (id) ON DELETE CASCADE NOT NULL,
+    name varchar(256) NOT NULL,
+    description text,
+    group_name varchar(256),
+    spec_data text,
+    spec_format event_api_spec_format,
+    spec_type event_api_spec_type,
+    version_value varchar(256),
+    version_deprecated bool,
+    version_deprecated_since varchar(256),
+    version_for_removal bool
+);
+
+CREATE INDEX ON event_api_definitions (tenant_id);
+
+-- Runtime Auth
+
+CREATE TABLE runtime_auths (
+    id uuid PRIMARY KEY,
+    tenant_id uuid NOT NULL,
+    runtime_id uuid REFERENCES runtimes (id) ON DELETE CASCADE NOT NULL,
+    app_def_id uuid REFERENCES api_definitions (id) ON DELETE CASCADE NOT NULL,
+    value jsonb
+);
+
+CREATE INDEX ON runtime_auths (tenant_id);
+CREATE UNIQUE INDEX ON runtime_auths (tenant_id, runtime_id, app_def_id);
+
+-- Document
+
+CREATE TYPE document_format AS ENUM (
+    'MARKDOWN'
+);
+
+CREATE TABLE documents (
+    id uuid PRIMARY KEY,
+    tenant_id uuid NOT NULL,
+    app_id uuid REFERENCES applications (id) ON DELETE CASCADE NOT NULL,
```

Comment:
please check if it is possible to validate that application belongs to the same tenant. If it is possible, apply it to the whole schema.